src/ooxml/java/org/apache/poi/xssf/usermodel/helpers/XSSFColumnShifter.java

```diff
@@ -0,0 +1,170 @@
+package org.apache.poi.xssf.usermodel.helpers;
+
+import java.util.ArrayList;
+import java.util.Comparator;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.SortedMap;
+import java.util.TreeMap;
+
+import org.apache.poi.ss.ITestDataProvider;
+import org.apache.poi.ss.formula.FormulaShifter;
+import org.apache.poi.ss.usermodel.Cell;
+import org.apache.poi.ss.usermodel.CellType;
+import org.apache.poi.ss.usermodel.Row;
+import org.apache.poi.ss.usermodel.Sheet;
+import org.apache.poi.ss.usermodel.helpers.ColumnShifter;
+import org.apache.poi.ss.util.CellRangeAddress;
+import org.apache.poi.ss.util.CellReference;
+import org.apache.poi.util.POILogFactory;
+import org.apache.poi.util.POILogger;
+import org.apache.poi.xssf.model.CommentsTable;
+import org.apache.poi.xssf.usermodel.XSSFCell;
+import org.apache.poi.xssf.usermodel.XSSFComment;
+import org.apache.poi.xssf.usermodel.XSSFRow;
+import org.apache.poi.xssf.usermodel.XSSFSheet;
+import org.apache.poi.xssf.usermodel.XSSFVMLDrawing;
+import org.openxmlformats.schemas.spreadsheetml.x2006.main.CTComment;
+import org.openxmlformats.schemas.spreadsheetml.x2006.main.CTCommentList;
+
+public class XSSFColumnShifter extends ColumnShifter{
+    
+    private static final POILogger logger = POILogFactory.getLogger(XSSFRowShifter.class);
+    
+    private int firstShiftColumnIndex; 
+    private int lastShiftColumnIndex; 
+    private int shiftStep;
+    
+    private XSSFShiftingManager formulaShiftingManager;
+
+   
+    public XSSFColumnShifter(Sheet sh, FormulaShifter shifter) {
+        super(sh, shifter);
+        formulaShiftingManager = new XSSFShiftingManager(sh, shifter);
+    }
+
+    public void shiftColumns(int firstShiftColumnIndex, int lastShiftColumnIndex, int step){
+        this.firstShiftColumnIndex = firstShiftColumnIndex;
+        this.lastShiftColumnIndex = lastShiftColumnIndex;
+        this.shiftStep = step;
+        if(shiftStep > 0)
```

Comment:
Are the variables `firstShiftColumnIndex`, `lastShiftColumnIndex`, and `shiftStep` used after `shiftColumns` is called? If not, these variables should be passed to `shiftColumnsRight` or `shiftColumnsLeft`, eliminating the need to store these variables. Eliminating unnecessary member variables makes the class easier to read (the class less closesly resembles a state machine) and slightly reduces the class's memory consumption.